### PR TITLE
Restore data sources

### PR DESCRIPTION
**Pull request by omadawn**
Added a filter so that we don't try and import data sources as dashboards (based entirely on filename)

Working restoration of datasource types. Similar filter based on file name. 

Nothing for users yet.

I am not 100% thrilled with the way I've implemented some of the functionality so I'm not really expecting this PR to be accepted as is but looking for input on it.

fixes issue #5

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -96,21 +96,27 @@ func main() {
 	case "backup":
 		// TODO fix logic accordingly with apply-for
 		doBackup(serverInstance, applyFor, matchDashboard)
+		fmt.Println("Backup Complete.")
```
Comment (grafov) [inline main.go]: Maybe better show these messages only in `verbose` mode? For CLI apps I try to stay with Unix ideology: utility should silently do it job and complain to stderr only on errors. So stdout should be used only for side effects (for example display listing) or in verbosity mode.

File: do-restore.go [hunk 2]
```diff
@@ -23,28 +23,137 @@ import (
 	"io/ioutil"
 	"os"
 	"strings"
+	"regexp"
+	"encoding/json"
+	"github.com/grafana-tools/sdk"
```
Comment (grafov) [inline do-restore.go]: Let split external import clauses from internal with a space line? It is not a strict requirement of course just minor style notify. I suppose it makes import statement more readable.

**Review by grafov — CHANGES_REQUESTED** [review]
I think this PR is good solution. I just have two minor remarks, see comments in the review.

File: do-restore.go (hunk 2 above)
Comment (omadawn) [reply do-restore.go]: _That's_ why I see so much code with a space in the imports. I was unaware of that convention. That makes sense.

File: main.go (hunk 1 above)
Comment (omadawn) [reply main.go]: Hrm... I'm two ways on that. You're right that a command line tool like this needs to play nicely with automation (that's pretty much exactly what it's for.) And all of the built in tools I can think of behave that way. However most of those all deal with local things and I've used a few of more complex tools which give you some indication by default.

That being said if we were to output a message by default we would want to add a --silent flag to make the thing shut up and that seems a little un-neccessary. I'll drop these messages and go back to the silent-by-default behavior when I update the PR for the above.

**Comment by omadawn** [comment]
PR Updated

**Comment by omadawn** [comment]
Of course now it's not restoring so I'll have to figure out what I broke and re-submit.

**Comment by grafov** [comment]
See the fixes in #11.

**Pull request closed without merging**